### PR TITLE
Incremental PR Analysis: Load unchanged changed files

### LINKED ISSUE #6483: Incremental PR Analysis: Load unchanged changed files
Part of https://sonarsource.atlassian.net/browse/MMF-2875

Load unchaged files generated by S4NET

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #6483

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Helpers/AnalysisConfig.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+namespace SonarAnalyzer.Helpers;
+
+/// <summary>
+/// Data class to describe an analysis configuration.
+/// </summary>
+/// <remarks>
+/// This class is the counterpart of SonarScanner.MSBuild.Common.AnalysisConfig, and is used for easy deserialization.
+/// This class should not be used in this codebase. To get configuration properties, use <see cref="AnalysisConfigReader"/>.
+/// </remarks>
+[XmlRoot(Namespace = "http://www.sonarsource.com/msbuild/integration/2015/1")]
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Helpers/AnalysisConfig.cs:32]: No, it's a serialization namespace. And there's no need to change it to HTTPS. That URL doesn't exist and will never be opened by anyone. XML Namespaces are just unique identifiers, that happen to have a form of URL.

File: analyzers/src/SonarAnalyzer.Common/Helpers/AnalysisConfigReader.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            using var stream = new FileStream(analysisConfigPath, FileMode.Open, FileAccess.Read);
+            analysisConfig = (AnalysisConfig)serializer.Deserialize(stream);
+        }
+        catch (Exception e)
+        {
+            throw new InvalidOperationException($"File '{analysisConfigPath}' could not be parsed.", e);
+        }
+    }
+
+    public string[] UnchangedFiles() =>
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Helpers/AnalysisConfigReader.cs:47]: Scanner is storing full paths. The .NET part of the analyzer is not supposed to care about the basepath problem.

See S4NET counterpart - this path is stored:
https://github.com/SonarSource/sonar-scanner-msbuild/blob/55ce2c76c4af7736d115420b56626284a9c62bd7/src/SonarScanner.MSBuild.PreProcessor/CacheProcessor.cs#L98

[the remark this answers is not in the document]
Comment (costin-zaharia-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Helpers/AnalysisConfigReader.cs:47]: Ah sry, mindfreeze :)

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**